vyxal/elements.py

```diff
@@ -2964,8 +2964,14 @@ def prepend(lhs, rhs, ctx):
 
 def product(lhs, ctx):
     """Element Π
-    (lst) -> product(list)"""
-    return vy_reduce(multiply, lhs, ctx=ctx)
+    (lst[num]) -> product(list)
+    (lst[str|lst]) -> Cartesian product over a list of lists
+    """
+
+    if all(vy_type(x) == NUMBER_TYPE for x in lhs):
```

Comment:
This will hang for infinite lists

Reply:
Neither product nor reduce by cartesian product make any sense for an infinite list. Doesn't matter.

Reply:
You right